### PR TITLE
StripSymbols default for PublishAOT

### LINKED ISSUE #35108: [Breaking change]: StripSymbols=true is now the default option for PublishAot
### Description

https://github.com/dotnet/runtime/pull/85139

When .NET 7 introduced Native AOT deployment, we exposed a `StripSymbols` property that optionally allows to strip debugging symbols from the produced executable on Linux into a separate file. The default value of the property was false.

In .NET 8 we're changing it to `true`.

### Version

.NET 8 (other)

### Previous behavior

Debugging symbols on Linux with `PublishAot` were placed into the produced executable by default with an opt-in to place them into a `.dbg` file.

### New behavior

Debugging symbols on Linux with `PublishAot` are placed into a `.dbg` file with an opt-out to place them into the executable.

### Type of breaking change

- [ ] **Binary incompatible**: Existing binaries may encounter a breaking change in behavior, such as failure to load or execute, and if so, require recompilation.
- [ ] **Source incompatible**: When recompiled using the new SDK or component or to target the new runtime, existing source code may require source changes to compile successfully.
- [X] **Behavioral change**: Existing binaries may behave differently at run time.

### Reason for change

Based on feedback, we established the .NET users prefer the .NET symbols convention instead of the platform-native convention. The option was not discoverable enough.

### Recommended action

* If you rely on debugging symbols being present in the main executable, add `<StripSymbols>false</StripSymbols>` to your project file to restore the previous behavior.
* If you choose to use the new default, verify the debugging symbols in .dbg files are properly archived if you expect you'll need to debug the generated executables.

### Feature area

Deployment

### Affected APIs

_No response_

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #35108.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/compatibility/8.0.md](https://github.com/dotnet/docs/blob/75126e53b2c8c9ee1a879aa43525e5181a01bfa8/docs/core/compatibility/8.0.md) | [Breaking changes in .NET 8](https://review.learn.microsoft.com/en-us/dotnet/core/compatibility/8.0?branch=pr-en-us-35229) |
| [docs/core/compatibility/deployment/8.0/stripsymbols-default.md](https://github.com/dotnet/docs/blob/75126e53b2c8c9ee1a879aa43525e5181a01bfa8/docs/core/compatibility/deployment/8.0/stripsymbols-default.md) | [docs/core/compatibility/deployment/8.0/stripsymbols-default](https://review.learn.microsoft.com/en-us/dotnet/core/compatibility/deployment/8.0/stripsymbols-default?branch=pr-en-us-35229) |
| [docs/core/deploying/native-aot/index.md](https://github.com/dotnet/docs/blob/75126e53b2c8c9ee1a879aa43525e5181a01bfa8/docs/core/deploying/native-aot/index.md) | [Native AOT deployment](https://review.learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/index?branch=pr-en-us-35229) |
| [docs/core/whats-new/dotnet-8.md](https://github.com/dotnet/docs/blob/75126e53b2c8c9ee1a879aa43525e5181a01bfa8/docs/core/whats-new/dotnet-8.md) | [Copy everything else and build app.](https://review.learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-8?branch=pr-en-us-35229) |
| [docs/standard/glossary.md](https://github.com/dotnet/docs/blob/75126e53b2c8c9ee1a879aa43525e5181a01bfa8/docs/standard/glossary.md) | [docs/standard/glossary](https://review.learn.microsoft.com/en-us/dotnet/standard/glossary?branch=pr-en-us-35229) |
| [docs/standard/native-interop/pinvoke-source-generation.md](https://github.com/dotnet/docs/blob/75126e53b2c8c9ee1a879aa43525e5181a01bfa8/docs/standard/native-interop/pinvoke-source-generation.md) | [Source generation for platform invokes](https://review.learn.microsoft.com/en-us/dotnet/standard/native-interop/pinvoke-source-generation?branch=pr-en-us-35229) |

### CODE REVIEW

File: docs/core/deploying/native-aot/index.md
Comment (MichalStrehovsky) [inline docs/core/deploying/native-aot/index.md:1]: On line 69 in this file there's a description of the old behavior of StripSymbols - would it be possible to make this into a section with content that depends on the selected version?

**Review by MichalStrehovsky — APPROVED** [review]
Thank you!

File: docs/core/deploying/native-aot/index.md
Comment (gewarren) [reply docs/core/deploying/native-aot/index.md:1]: @MichalStrehovsky Does this look okay?

https://review.learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/?branch=pr-en-us-35229&tabs=net7#native-debug-information

File: docs/core/deploying/native-aot/index.md [hunk 1]
```diff
@@ -64,35 +64,49 @@ On Linux, install compiler toolchain and developer packages for libraries that .
...
 
 ```xml
 <PropertyGroup>
     <StripSymbols>true</StripSymbols>
 </PropertyGroup>
 ```
 
+### [.NET 8+](#tab/net8plus)
+
+By default, native AOT publishing produces debug information in a separate *.pdb* file on both Windows and Linux. On Linux only, you can set the `StripSymbols` property to `false` to include the debug information in the native binary. (Including the debug information makes the native binary significantly larger.)
```
Comment (MichalStrehovsky) [inline docs/core/deploying/native-aot/index.md:85]:
```suggestion
By default, native AOT publishing produces debug information in a separate file on all supported platforms (*.dbg* on Linux, *.pdb* on Windows, and *.dwarf* on macOS). This file is necessary for running the app under debugger or inspecting crash dumps. On Unix-like platforms, you can set the `StripSymbols` property to `false` to include the debug information in the native binary. (Including the debug information makes the native binary significantly larger.)
```

**Review by MichalStrehovsky — APPROVED** [review]
One suggestion - we're adding macOS support in .NET 8.

**Pull request merged**